packages/headless/src/features/insight-search/insight-search-request.ts

```diff
@@ -50,7 +50,19 @@ export const buildInsightBaseRequest = async (
         fieldsToInclude: state.fields.fieldsToInclude,
       }),
     ...(state.didYouMean && {
-      enableDidYouMean: state.didYouMean.enableDidYouMean,
+      queryCorrection: {
+        enabled:
+          state.didYouMean.enableDidYouMean &&
+          state.didYouMean.queryCorrectionMode === 'next',
+        options: {
+          automaticallyCorrect: state.didYouMean.automaticallyCorrectQuery
+            ? ('whenNoResults' as const)
+            : ('never' as const),
+        },
+      },
```

Comment:
I think all this object needs to be set only when working on `Next`.
like `enableDidYouMean` needs to be set only on `legacy`